WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationDetailsViewController.m

```diff
@@ -518,28 +518,37 @@ - (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath
 
 #pragma mark - Media Download Helper
 
-- (void)enqueueMediaDownloads:(NotificationBlockGroup *)group indexPath:(NSIndexPath *)indexPath
+- (void)downloadAndResizeMedia:(NSIndexPath *)indexPath blockGroup:(NotificationBlockGroup *)blockGroup
 {
-    // Download Media: Only embeds for Text and Comment notifications
+    //  Notes:
+    //  -   We'll *only* download Media for Text and Comment Blocks
+    //  -   Plus, we'll also resize the downloaded media cache *if needed*. This is meant to adjust images to
+    //      better fit onscreen, whenever the device orientation changes (and in turn, the maxMediaEmbedWidth changes too).
+    //
     NSSet *richBlockTypes           = [NSSet setWithObjects:@(NoteBlockTypeText), @(NoteBlockTypeComment), nil];
-    NSSet *imageUrls                = [group imageUrlsForBlocksOfTypes:richBlockTypes];
+    NSSet *imageUrls                = [blockGroup imageUrlsForBlocksOfTypes:richBlockTypes];
     __weak __typeof(self) weakSelf  = self;
     
-    [self.mediaDownloader downloadMediaWithUrls:imageUrls completion:^{
-        [weakSelf.tableView reloadRowsAtIndexPaths:@[indexPath] withRowAnimation:UITableViewRowAnimationFade];
-    }];
+    void (^completion)(void)        = ^{
+        
+        // Workaround:
+        // Performing the reload call, multiple times, without the UIViewAnimationOptionBeginFromCurrentState might lead
+        // to a state in which the cell remains not visible.
+        //
+        [UIView animateWithDuration:0.25 delay:0.0 options:UIViewAnimationOptionOverrideInheritedDuration | UIViewAnimationOptionBeginFromCurrentState animations:^{
+            [weakSelf.tableView reloadRowsAtIndexPaths:@[indexPath] withRowAnimation:UITableViewRowAnimationFade];
+        } completion:nil];
+    };
+
+    [self.mediaDownloader downloadMediaWithUrls:imageUrls maximumWidth:self.maxMediaEmbedWidth completion:completion];
+    [self.mediaDownloader resizeMediaWithIncorrectSize:self.maxMediaEmbedWidth completion:completion];
 }
 
 - (CGFloat)maxMediaEmbedWidth
 {
-    // Maximum Media Embed Width should match with the RichTextView's width in portrait mode
-    NoteBlockTextTableViewCell *textCell = self.layoutCellMap[@(NoteBlockGroupTypeText)];
-    NSAssert(textCell, @"Missing TextCell?");
-    
-    // Note: First iteration doesn't support embed resize on rotation. Let's take the portrait width
-    CGRect bounds           = self.view.bounds;
-    CGFloat portraitWidth   = [UIDevice isPad] ? WPTableViewFixedWidth : MIN(CGRectGetWidth(bounds), CGRectGetHeight(bounds));
-    CGFloat maxWidth        = portraitWidth - textCell.labelPadding.left - textCell.labelPadding.right;
+    UIEdgeInsets textPadding        = NoteBlockTextTableViewCell.defaultLabelPadding;
+    CGFloat portraitWidth           = [UIDevice isPad] ? WPTableViewFixedWidth : CGRectGetWidth(self.view.bounds);
+    CGFloat maxWidth                = portraitWidth - textPadding.left - textPadding.right;
```

Comment:
Sure, no problem! updating!. I'm just used to writing it this way, precisely, since it's left to right, and the left padding won't get the right substracted. But i think you're right, and it'll be clearer!